api/tests/acceptance/application/session/session-controller-patch_test.js

```diff
@@ -76,21 +76,6 @@ describe('Acceptance | Controller | session-controller-patch', () => {
         });
     });
 
-    it('should return 404 HTTP status code when session cannot be found', () => {
-      // when
-      const promise = server.inject({
-        method: 'PATCH',
-        url: '/api/sessions/2',
-        headers: { authorization: generateValidRequestAuthorizationHeader(user.id) },
-        payload
-      });
-
-      // then
-      return promise.then((response) => {
-        expect(response.statusCode).to.equal(404);
-      });
-    });
-
     it('should respond with a 403 when user is not authorized to update the session', function() {
```

Comment:
Je ne trouve pas ça super sécurisé de répondre avec une 403. Ça donne trop d'indice je pense sur l'existence de cette ressource. 

Je me suis pas mal renseigné ce week-end sur "vaut-il mieux renvoyer une 403, une 404 ou une 400".

La réponse que j'ai actuellement c'est que si la ressource est "critique", et qu'on ne veut pas donner d'indication au client, dans le cas d'une lecture, on peut renvoyer une 404, indifféremment du cas qui se produit (Resource Not Found ou Forbidden), avec un message d'erreur du style : "La ressource n'existe pas ou son accès est restreint."

Si la ressource est censée être publique (ex : un profil LinkedIn, GitHub ou Twitter), on peut dissocier les 2 cas.

Reply:
Ok je vais modifier ce morceau quand j'ai un moment.

Reply:
OK donc là si je comprends bien, quand la session n'existe pas on a une réponse du genre `404 La session n'existe pas` mais si on n'est pas autorisé à la voir on a `404 La session n'existe pas ou son accès est restreint` ? Si c'est ça on n'y est pas encore, l'idée c'est que le client ne puisse _pas_ faire la différence 😉